MetaMask Plugin/Tutorial.md

```diff
@@ -0,0 +1,244 @@
+
+* Click the "Connect" button on the site.
+* Approve the site's permissions request (which includes the Snap installation!)
+* Approve the Snap's permissions request (which in this case is permission to show alerts to you,   to send its message)
+* Click the "Send Hello" button to receive a greeting from the Snap.
+
+
+
+### Abstract level code description.
+
+File Structure :- 
+Dist
+Bundle.js - bundled js file of snap
+Index.js - snap code
+Package.json - Permissions are put here
+Package.-lock.json
+Snap.config.json - Snap configuration 
+ 
+The permissions the Snap initially requests are configured in its `package.json`, under the `web3Wallet key`.
+
+`index.html` file, interacts with the Snap using two basic API calls. It contains the DApp code. 
+In the `index.js` file, you can add API methods to connected websites from within a Snap. It contains the Snap Code.
+
+#### Snap Code `index.js`
+```
+wallet.registerRpcMessageHandler(async (originString, requestObject) => {
+  switch (requestObject.method) {
+    case 'hello':
+      return wallet.send({
+        method: 'alert',
+        params: [`Hello, ${originString}!`]
+      })
+    default:
+      throw new Error('Method not found.')
+  }
+})
+```
+
+The code registers a RPC Handler i.e., creates a developer defined API by the name of `hello` which can be called via the frontend given below. 
+
+- `requestObject` contains the method to be executed
+- `Alert` is an inbuilt method which allows us to create an alert on the webpage. 
+- An alert is created when the hello method is called using the metamask api.
+
+#### Dapp Code :- `index.html`
+```
+async function connect () {
+      await ethereum.send({
+        method: 'wallet_enable',
+        params: [{
+          wallet_plugin: { [snapId]: {} },
+        }]
+      })
+    }
+
+    // here we call the plugin's "hello" method
+ async function send () {
+      try {
+        const response = await ethereum.send({
+          method: 'wallet_invokePlugin',
+          params: [snapId, {
+            method: 'hello'
+          }]
+        })
+      } catch (err) {
+        console.error(err)
+        alert('Problem happened: ' + err.message || err)
+      }
+    }
+```
+- `Connect` function is called to connect metamask with the plugin and download it if does not exist. 
+- `wallet_enable` - when this method is sent, the metamask asks user for the permissions of the plugin.
+- `wallet_invokePlugin` is another method used to call an RPC method we declared above `hello` in our case. The hello case in our switch statement is called leading to an alert.
+
+**Note**: For starting your own Snap, you might want to just copy one of the examples to get started! Then you can follow up by using plugin APIs explained below and make edits in `index.html` and `index.js`.
+
+
+### Plugin APIs
+
+APIs Currently Provided -
```

Comment:
@thesachinmittal Most of this is just copy and pasted from the docs, so probably better to not replicate that content (which will update quickly) and just link to it somewhere appropriate.

Reply:
@ChrisChinchilla Truly said, but I felt if we provide the link, reader will have to navigate through the repo... Instead they can follow this walkthrough with Screenshots.

Reply:
Yes, but if those docs change, which they will, as this is a beta release, then this article is out of date. And there aren't any screenshots in this section anyway, or any necessary steps for the tutorial, as far as I can tell.

How about as a compromise, you highlight the methods that this code example uses, and then link to the documentation to tell people where to find the rest?

Reply:
Indeed @ChrisChinchilla , I have already highlighted and explained the methods, I will provide the link for the walkthrough...

Reply:
@ChrisChinchilla , I have added the screenshots because when I was building the plugin. It takes time to debug and follow through the tutorial given on metamask plugin. 

My Plugin: https://www.loom.com/share/fa9ab07c7dbc48f497b0c866fc5faeab